source/math/RandomGenerator.ooc

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright (c) 2011-2014 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+import math
+use ooc-collections
+import os/Time
+
+RandomGenerator: abstract class {
+	_state: Int
+	init: func {
+		this _state = Time microtime()
+	}
+	init: func ~withSeed (=_state)
+	next: abstract func -> Float
+	next: func ~withCount (count: Int) -> Float[] {
+		result := Float[count] new()
+		for (i in 0 .. count)
+			result[i] = this next()
+		result
+	}
+	_next: func -> Float {
+		this _state = (214013 * this _state) + 2531011
+		((((this _state >> 16) & 0x7fff) % 10000) as Float) / 10000.0f
+	}
+}
+
+UniformRandomGenerator: class extends RandomGenerator {
+	_min := 0.0f
+	_max := 1.0f
+	init: func {
+		super()
+	}
+	init: func ~withSeed (seed: Int) {
```

Comment:
You could also replace this function with
```ooc
init: func ~withSeed (seed := Time microtime(), min := 0.0f, max := 1.0f) {
   super(seed)
   this _min = min
   this _max = max
}
```
so that the order of the parameters is always the same. Downside is, you *could* call the function with a specified `min` parameter and leave out the `max` one, which would be weird.